clang/lib/Basic/Targets/AArch64.cpp

```diff
@@ -1480,11 +1480,11 @@ AArch64leTargetInfo::AArch64leTargetInfo(const llvm::Triple &Triple,
 void AArch64leTargetInfo::setDataLayout() {
   if (getTriple().isOSBinFormatMachO()) {
     if(getTriple().isArch32Bit())
-      resetDataLayout("e-m:o-p:32:32-i64:64-i128:128-n32:64-S128", "_");
+      resetDataLayout("e-m:o-p:32:32-i64:64-i128:128-n32:64-S128-Fn32", "_");
```

Comment:
Just to double check @TNorthover, is an alignment of 32 OK for arm64_32?

Reply:
Yep. CPU would hard-fault if that was violated. All instructions are 4 bytes and must be aligned.

Reply:
Is this true even if the cpu was executing thumb2 instructions in AArch32 mode?

Reply:
Thumb2 function pointers aren't aligned... but from LLVM's perspective it's a completely different target, so it's not relevant.